app/components/elements/button_component.rb

```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+
+module Elements
+  # Component for a button
+  class ButtonComponent < ApplicationComponent
+    def initialize(label:, classes: [], variant: nil, size: nil, **options)
+      @label = label
+      @classes = classes
+      @variant = variant
+      @size = size
+      @options = options
+      super()
+    end
+
+    attr_reader :label, :options
+
+    def call
+      tag.button(
+        class: Elements::ButtonSupport.classes(variant: @variant, size: @size, classes:),
+        type: 'button',
+        **options
+      ) do
+        label
+      end
+    end
+
+    def classes
+      merge_classes(@classes)
+    end
+  end
+end
```

Comment:
Sometimes we need a button component that isn't a link, e.g., for initiating Stimulus actions, which is part of making repeatable form fields.